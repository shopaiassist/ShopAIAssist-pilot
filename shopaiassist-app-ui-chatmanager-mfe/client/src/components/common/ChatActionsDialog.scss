.chat-actions-dialog {
   .chat-actions-dialog-body {
      min-width: 35rem;
      display: flex;
      flex-direction: column;
      gap: 1rem;
      padding-bottom: 0.25rem;
   }
   .chat-rename-field {
      display: flex;
      flex-direction: column;
      gap: 1rem;
   }
}
