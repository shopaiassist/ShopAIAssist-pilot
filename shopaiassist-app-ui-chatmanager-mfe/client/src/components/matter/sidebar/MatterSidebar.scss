@import "../../../utils/styles/imports";

$headerHeight: 3.75rem;

.matter-sidebar-container {
   display: flex;
   width: 21rem;
   z-index: map-get($z-index-map, "default");

   &.compact-mode {
      position:absolute;
      width: 100%;
   }

   &.full-mode {
      position:relative;
      width: 21rem;
      max-width: 100%;
   }

   .matter-sidebar-nav-header {
      display: flex;
      padding: 0.5rem;
      justify-content: space-between;
      align-items: center;
      align-self: stretch;
      min-height: 3.688rem;
      box-sizing: border-box;
      width: 100%;
      border: rem-calc(1);
      border-bottom: $saf-line-width-thin solid $saf-color-line-subtle;
      border-left: $saf-line-width-thin solid $saf-color-line-subtle;

      .matter-sidebar-nav-back {
         display: flex;
         gap: 0.5rem;
      }
   }

   .matter-sidebar {
      display: flex;
      flex-direction: column;
      align-items: center;
      box-shadow: $saf-drop-shadow-level-3;
      height: calc(100vh - $headerHeight);
      width: 21rem;
      background-color: $saf-color-background-subtle;

      &.compact-mode {
         width: 100%;
      }

      &.full-mode {
         width: 21rem;
         max-width: 100%;
      }

      .matter-sidebar-folder-name {
         padding: 0.5rem;
         margin: 0.75rem;
         display: flex;
         align-self: flex-start;
      }

      .matter-sidebar-header {
         display: flex;
         flex-direction: row;
         justify-content: space-between;
         width: 95%;
         padding: 0.5rem;
      }

      .matter-sidebar-footer {
         display: flex;
         width: 100%;
      }
   }
}