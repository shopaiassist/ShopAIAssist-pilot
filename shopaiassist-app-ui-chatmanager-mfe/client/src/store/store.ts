import { Action, AsyncThunkPayloadCreator, configureStore, ThunkAction } from '@reduxjs/toolkit';

import userReducer from './userSlice';

export const reducer = {
  user: userReducer
};

export const store = configureStore({
  reducer
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;

export interface AppThunkApiConfig {
  dispatch: AppDispatch;
  rejectValue: string;
  state: RootState;
}
export type AppAsyncThunk<P, R> = AsyncThunkPayloadCreator<R, P, AppThunkApiConfig>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>;
