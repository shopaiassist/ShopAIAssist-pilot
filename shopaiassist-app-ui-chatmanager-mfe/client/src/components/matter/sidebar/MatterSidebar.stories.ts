// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
import MatterSidebar from './MatterSidebar';
import { Meta, StoryObj } from '@storybook/react';

const meta = {
  title: 'Matter/Sidebar',
  component: MatterSidebar,
  parameters: {
    layout: 'padded'
  },
  tags: ['autodocs']
} satisfies Meta<typeof MatterSidebar>;

export default meta;

export const Default: StoryObj = { args: {} };
