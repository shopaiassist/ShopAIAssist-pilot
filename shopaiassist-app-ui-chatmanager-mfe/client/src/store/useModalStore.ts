import { create } from 'zustand';
import { devtools } from 'zustand/middleware';
import { FolderItem, TreeItem } from '../@types/sidebar';

export enum ChatDialogState {
  DELETE_CHAT = 'delete',
  UPDATE_CHAT = 'update'
}

/**
 * Represents the state structure managed in the store for modal interactions.
 * This interface includes state management for items that could trigger modal actions,
 * and controls for showing or hiding the modal.
 *
 * @interface ModalState
 * @property {(TreeItem|FolderItem)|undefined} actionItem - The current item (chat or folder) selected for actions like edit/delete.
 * @property {boolean} hideModal - A boolean flag to indicate if the modal should be hidden or shown.
 * @property {ChatDialogState|undefined} modalState - The current state of the modal (e.g., delete, update).
 * @property {Function} setActionItem - Sets the current item for modal actions.
 * @property {Function} setModalState - Sets the current state of the modal.
 * @property {Function} toggleModal - Toggles the modal's visibility state.
 */
export interface ModalState {
  actionItem?: TreeItem | FolderItem;
  modalState?: ChatDialogState;
  hideModal: boolean;
  setActionItem: (newActionItem?: TreeItem | FolderItem) => void;
  setModalState: (newState?: ChatDialogState) => void;
  toggleModal: () => void;
  dismiss: () => void;
}

/**
 * Creates a Zustand store to manage modal state using Zustand along with devtools for debugging.
 * The store contains state and actions related to modal interactions for items such as TreeItem and FolderItem.
 *
 * @returns {UseStore<ModalState>} A Zustand store with the following:
 *         - `actionItem`: The item currently targeted for modal actions.
 *         - `hideModal`: Boolean indicating if the modal is currently shown or hidden.
 *         - `modalState`: The current state of the modal (e.g., delete, update).
 *         - `setActionItem`: Function to set the item to be used in modal actions.
 *         - `setModalState`: Function to set the current state of the modal.
 *         - `toggleModal`: Function to toggle the visibility of the modal.
 */

export const useModalState = create<ModalState>()(
  devtools(
    (set, get) => ({
      actionItem: undefined,
      hideModal: true,
      modalState: undefined,
      setActionItem: (newActionItem?: TreeItem | FolderItem) => {
        set({ actionItem: newActionItem });
      },
      setModalState: (newState?: ChatDialogState) => {
        set({ modalState: newState });
      },
      toggleModal: () => {
        set({ hideModal: !get().hideModal });
      },
      dismiss: () => {
        set({ hideModal: true });
      }
    }),
    { name: 'useModalStore' }
  )
);
