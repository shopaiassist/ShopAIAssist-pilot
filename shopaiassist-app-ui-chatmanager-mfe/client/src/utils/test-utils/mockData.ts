import { TreeItem, FolderItem } from '../../@types/sidebar';

export const mockTree: (TreeItem | FolderItem)[] = [
  {
    treeItemId: '9a4144e0-65ea-4f6c-8cd5-43a0a6d1e634',
    name: 'Test Folder For Created At 2',
    type: 'folder',
    createdAt: new Date('2024-05-06T15:26:57.992Z'),
    updatedAt: new Date('2024-05-06T15:26:57.992Z'),
    fileCollectionId: 'whatever'
  },
  {
    treeItemId: '5aa13d2b-3f04-4a91-a84b-bf214420b47c',
    name: 'Test Folder For Created At 2',
    type: 'folder',
    createdAt: new Date('2024-05-06T15:19:07.430Z'),
    updatedAt: new Date('2024-05-06T15:19:07.430Z'),
    fileCollectionId: 'whatever'
  },
  {
    type: 'folder',
    name: 'New Folder 100',
    description: 'This folder is used for whatever, its created via postman.',
    matterId: 'Whatever Folder 100',
    treeItemId: '4c484fb3-faf6-4fe8-9930-410772f141d7',
    createdAt: new Date('2024-04-25T19:21:07.787Z'),
    updatedAt: new Date('2024-04-25T19:21:07.787Z'),
    fileCollectionId: 'whatever'
  },
  {
    createdAt: new Date('2024-03-01T17:00:33.564Z'),
    treeItemId: 'e35607c7-be8f-4172-a3cd-ae7e39bc2b96',
    name: 'A chat named desire',
    type: 'chat',
    updatedAt: new Date('2024-05-06T18:09:49.245Z'),
    fileCollectionId: 'whatever'
  },
  {
    createdAt: new Date('2024-02-23T16:30:33.080Z'),
    updatedAt: new Date('2024-02-23T16:30:33.080Z'),
    treeItemId: '8c1b78b3-a798-4557-871a-bff6c8ac804e',
    name: 'new2',
    type: 'folder',
    fileCollectionId: 'whatever'
  },
  {
    createdAt: new Date('2024-02-23T16:30:17.078Z'),
    updatedAt: new Date('2024-02-23T16:30:17.078Z'),
    treeItemId: 'b39f37a3-6781-4445-8573-77658d9ee7ae',
    name: 'new1',
    type: 'folder',
    fileCollectionId: 'whatever'
  },
  {
    createdAt: new Date('2024-02-23T16:26:17.254Z'),
    updatedAt: new Date('2024-02-23T16:26:17.254Z'),
    treeItemId: '9bc3c583-a6e4-41cc-92f1-27e1be3ba4f6',
    name: 'newnewenewnew',
    type: 'folder',
    fileCollectionId: 'whatever'
  },
  {
    createdAt: new Date('2024-02-23T16:25:50.258Z'),
    updatedAt: new Date('2024-02-23T16:25:50.258Z'),
    treeItemId: '2ec6bba4-6dd0-4190-a015-b58e6ee4021c',
    name: '12346123445',
    type: 'folder',
    fileCollectionId: 'whatever'
  },
  {
    createdAt: new Date('2024-02-21T20:51:41.137Z'),
    updatedAt: new Date('2024-02-21T20:51:41.137Z'),
    treeItemId: 'e4a9bc6a-cfb1-48ac-bc6e-1b6f135147c0',
    name: 'Chat Name 2',
    type: 'chat',
    fileCollectionId: 'whatever'
  },
  {
    createdAt: new Date('2024-02-20T19:47:03.738Z'),
    updatedAt: new Date('2024-02-20T19:47:03.738Z'),
    treeItemId: 'ce2422ea-b573-4c71-b569-fa2c79a5e50b',
    name: 'B matter with B name',
    type: 'folder',
    fileCollectionId: 'whatever'
  },
  {
    createdAt: new Date('2024-02-20T19:44:31.642Z'),
    updatedAt: new Date('2024-02-20T19:44:31.642Z'),
    treeItemId: '148f72ac-b903-451c-88f2-ed1282c6acb1',
    name: 'Zmatter with Z name',
    matterId: 'A matter with A name',
    type: 'folder',
    fileCollectionId: 'whatever'
  },
  {
    createdAt: new Date('2024-02-15T18:35:41.269Z'),
    updatedAt: new Date('2024-02-15T18:35:41.269Z'),
    treeItemId: 'e82e81e6-68c8-4bd1-b67c-316dc4105d18',
    name: '123',
    matterId: 'Matter with a really REALLY long name, im mean like REALLY long',
    type: 'folder',
    fileCollectionId: 'whatever'
  }
];
