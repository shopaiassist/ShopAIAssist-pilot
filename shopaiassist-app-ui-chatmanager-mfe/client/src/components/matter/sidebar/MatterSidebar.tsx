import { Text } from '@on/core-components/react';
import MatterSidebarHeader from './header/MatterSidebarHeader';
import ChatList from './chat-list/ChatList';
import { FolderItem, TreeItem } from '../../../@types/sidebar';
import MatterSidebarNavigation from './header/MatterSidebarNavigation';
import useMeasure from 'react-use-measure';
import { animated, useSpring } from '@react-spring/web';
import { useChatSidebarStore } from '../../../store/useChatSidebarStore';
import MatterSidebarFooter from './footer/MatterSidebarFooter';

import './MatterSidebar.scss';
import classNames from 'classnames';
import { ModalType } from '../../../App';
import { useMatters } from '../../../views/Context';

interface MatterSidebarProps {
  isOpen: boolean;
  tree?: (TreeItem | FolderItem)[];
  onNewChat: () => void;
  closeSidebar: () => void;
}

/**
 * A responsive sidebar component for displaying a tree of items such as chats and folders.
 * This sidebar can be toggled to show or hide using an animated transition.
 * The component adapts to changes in the sidebar's width and employs animations for smooth transitions.
 *
 * @param {MatterSidebarProps} props - The properties for the MatterSidebar component.
 * @param {(TreeItem | FolderItem)[]} [props.tree] - Data structure representing the tree items to be displayed in the sidebar.
 * @param {() => void} props.onNewChat - Callback function that triggers when a new chat is initiated.
 * @returns {JSX.Element} The animated sidebar containing a header and a list of chat or folder items.
 *
 * @example
 * <MatterSidebar tree={treeData} onNewChat={handleNewChat} />
 */
const MatterSidebar = ({ isOpen, tree, onNewChat, closeSidebar }: MatterSidebarProps) => {
  const [measureRef, { width }] = useMeasure();
  const springProps = useSpring({
    transform: !isOpen ? `translateX(-${width}px)` : 'translateX(0px)',
    marginRight: !isOpen ? `-${width}px` : '0px'
  });
  const [
    activeFolder,
    setIsMattersettingsActive,
    setIsMatterFilesActive,
    setActiveChat,
    archivedMattersView,
    toggleArchivedMattersView
  ] = useChatSidebarStore((state) => [
    state.activeFolder,
    state.setIsMattersettingsActive,
    state.setIsMatterFilesActive,
    state.setActiveChat,
    state.archivedMattersView,
    state.toggleArchivedMattersView
  ]);
  const { modalType } = useMatters();

  return (
    <animated.div
      style={springProps}
      className={classNames('matter-sidebar-container', {
        'compact-mode': modalType === ModalType.Compact,
        'full-mode': modalType !== ModalType.Compact
      })}
    >
      <div
        ref={measureRef}
        className={classNames('matter-sidebar', {
          'compact-mode': modalType === ModalType.Compact,
          'full-mode': modalType !== ModalType.Compact
        })}
        data-testid="matter-sidebar"
        is-open={isOpen.toString()}
      >
        <MatterSidebarNavigation
          className="matter-sidebar-nav-header"
          isSubsectionView={!!activeFolder || !!archivedMattersView}
          closeSidebar={closeSidebar}
        />
        {activeFolder && (
          <div className="matter-sidebar-folder-name">
            <Text appearance="heading-lg">{activeFolder?.name}</Text>
          </div>
        )}
        {archivedMattersView && (
          <div className="matter-sidebar-folder-name">
            <Text appearance="heading-lg">Archived Matters</Text>
          </div>
        )}
        <MatterSidebarHeader
          className="matter-sidebar-header"
          onNewChat={() => {
            onNewChat();
            if (modalType === ModalType.Compact) {
              closeSidebar();
            }
          }}
          showAddChatButton={!archivedMattersView}
          isFolderView={!!activeFolder}
        />
        <ChatList chatsAndFolders={tree} closeSidebar={closeSidebar} />

        {/* <MatterSidebarFooter
          activeFolder={activeFolder}
          archivedMattersView={archivedMattersView}
          onOpenSettings={() => {
            setIsMattersettingsActive(true), setIsMatterFilesActive(false);
          }}
          onOpenMatterFiles={() => {
            setActiveChat();
            setIsMatterFilesActive(true);
          }}
          toggleArchivedMattersView={toggleArchivedMattersView}
        /> */}
      </div>
    </animated.div>
  );
};

export default MatterSidebar;
