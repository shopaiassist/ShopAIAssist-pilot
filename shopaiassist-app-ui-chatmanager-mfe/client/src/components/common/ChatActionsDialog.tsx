import { Alert, Button, Dialog, Text, Textfield } from '@on/core-components/react';
import Translate from '../i18n/Translate';
import { useTranslation } from 'react-i18next';
import useChatActionsDialog from '../../hooks/useChatActionsDialog';
import { ChatDialogState } from '../../store/useModalStore';

import './ChatActionsDialog.scss';

/**
 * ChatActionsDialog component handles the UI for various chat actions such as renaming and deleting a chat.
 * It uses Dialog for the modal, Textfield for input, and Button for actions.
 *
 * The component dynamically switches between different states (like deleting a chat or renaming it)
 * and updates its content accordingly using the state and actions from the `useChatActionsDialog` hook.
 *
 * @component
 * @returns {JSX.Element} The rendered ChatActionsDialog component.
 */
const ChatActionsDialog = () => {
  const [dialogCopy, chatName, hideModal, modalState, onUpdateChatName, onCancel, onModalAction, dismiss] =
    useChatActionsDialog();
  const { t } = useTranslation();
  return (
    <Dialog
      hidden={hideModal}
      dialogTitle={t(dialogCopy.title)}
      isFooter="true"
      onClose={onCancel}
      onHide={dismiss}
      className="chat-actions-dialog"
    >
      <div className="chat-actions-dialog-body">
        {modalState === ChatDialogState.DELETE_CHAT ? (
          <Alert appearance="error" hideCloseButton={true}>
            <Translate tKey="CONFIRM_CHAT_DELETE" />
          </Alert>
        ) : (
          <div className="chat-rename-field">
            <Text appearance="heading-md">{`* ${t('INDICATES_REQUIRED_FIELD')}`}</Text>
            <Textfield
              data-testid="chat-name-field"
              label={`${t('NAME')}*`}
              maxlength={128}
              value={chatName}
              onInput={(e) => {
                onUpdateChatName(e.target.value);
              }}
            />
          </div>
        )}
      </div>
      <div slot="footer" className="delete-dialog-footer">
        <Button data-testid="chat-dialog-cancel" appearance="secondary" onClick={onCancel}>
          <Translate tKey={'CANCEL'} />
        </Button>
        <Button
          data-testid="chat-dialog-action"
          appearance="primary"
          disabled={!chatName?.length}
          onClick={onModalAction}
        >
          <Translate tKey={dialogCopy.actionButton} />
        </Button>
      </div>
    </Dialog >
  );
};
export default ChatActionsDialog;
