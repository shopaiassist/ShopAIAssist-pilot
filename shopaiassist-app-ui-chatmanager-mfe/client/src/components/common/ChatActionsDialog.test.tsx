import axios from 'axios';
import { act, fireEvent, renderWithI18n, snapshot, waitFor } from '../../../jest-utils';
import { TreeItem } from '../../@types/sidebar';
import { ChatDialogState, useModalState } from '../../store/useModalStore';
import ChatActionsDialog from './ChatActionsDialog';
import { mockTree } from '../../utils/test-utils/mockData';

const getMockedComponent = () => {
  return <ChatActionsDialog />;
};

jest.mock('axios');
const mockedAxios = axios as jest.Mocked<typeof axios>;
const mockedChat: TreeItem = {
  treeItemId: 'fake-chat-id',
  name: 'fakeChat',
  createdAt: new Date(),
  updatedAt: new Date(),
  type: 'chat',
  fileCollectionId: 'whatever'
};

describe('ChatActionsDialog Tests', () => {
  const store = useModalState.getState();

  beforeEach(() => {
    useModalState.setState(store, true);
    jest.clearAllMocks();
  });

  describe('render', () => {
    it('should render update chat name', async () => {
      store.setActionItem(mockedChat);
      store.setModalState(ChatDialogState.UPDATE_CHAT);
      store.toggleModal();
      snapshot(getMockedComponent());
    });
    it('should render delete chat', async () => {
      store.setActionItem(mockedChat);
      store.setModalState(ChatDialogState.DELETE_CHAT);
      store.toggleModal();
      snapshot(getMockedComponent());
    });
  });

  describe('user action', () => {
    it('should call to delete a chat', async () => {
      store.setActionItem(mockedChat);
      store.setModalState(ChatDialogState.DELETE_CHAT);
      store.toggleModal();
      mockedAxios.get.mockResolvedValue(mockTree);
      mockedAxios.delete.mockResolvedValue({ status: 200 });
      const component = getMockedComponent();
      const wrapper = renderWithI18n(component);

      const confirm = wrapper.getByTestId('chat-dialog-action');
      confirm.click();

      await act(() => waitFor(() => wrapper.rerender(component)));
      expect(mockedAxios.delete).toHaveBeenCalledWith('jest-test-domain/api/chats', {
        data: { id: mockedChat.treeItemId }
      });
    });

    it('should call to update a matter', async () => {
      store.setActionItem(mockedChat);
      store.setModalState(ChatDialogState.UPDATE_CHAT);
      store.toggleModal();
      mockedAxios.get.mockResolvedValue(mockTree);
      mockedAxios.patch.mockResolvedValue({ status: 200 });
      const component = getMockedComponent();
      const wrapper = renderWithI18n(component);
      const confirm = wrapper.getByTestId('chat-dialog-action');
      expect(confirm).not.toBeDisabled();

      const nameField = wrapper.getByTestId('chat-name-field');
      expect(nameField).not.toBeDisabled();
      fireEvent.input(nameField, { target: { value: '' } });
      expect(confirm).toBeDisabled();
      fireEvent.input(nameField, { target: { value: 'Changed Name' } });

      confirm.click();
      await act(() => waitFor(() => wrapper.rerender(component)));
      expect(mockedAxios.patch).toHaveBeenCalledWith('jest-test-domain/api/chats', {
        id: mockedChat.treeItemId,
        updates: { name: 'Changed Name' }
      });
    });

    it('should close when user clicks cancel', async () => {
      store.setActionItem(mockedChat);
      store.setModalState(ChatDialogState.DELETE_CHAT);
      store.toggleModal();
      const component = getMockedComponent();
      const wrapper = renderWithI18n(component);
      const confirm = wrapper.getByTestId('chat-dialog-cancel');
      confirm.click();
      await act(() => waitFor(() => wrapper.rerender(component)));
      expect(store.hideModal).toBe(true);
    });
  });
});
