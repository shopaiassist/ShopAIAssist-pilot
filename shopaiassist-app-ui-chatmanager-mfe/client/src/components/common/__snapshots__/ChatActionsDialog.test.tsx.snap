// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatActionsDialog Tests render should render delete chat 1`] = `
<div
  class="chat-actions-dialog"
  dialogtitle="Delete chat?"
  isfooter="true"
>
  <div
    class="chat-actions-dialog-body"
  >
    <span
      appearance="error"
    >
      Any files within this chat will be lost. This canâ€™t be undone.
    </span>
  </div>
  <div
    class="delete-dialog-footer"
    slot="footer"
  >
    <button
      appearance="secondary"
      data-testid="chat-dialog-cancel"
    >
      Cancel
    </button>
    <button
      appearance="primary"
      data-testid="chat-dialog-action"
    >
      Delete chat
    </button>
  </div>
</div>
`;

exports[`ChatActionsDialog Tests render should render update chat name 1`] = `
<div
  class="chat-actions-dialog"
  dialogtitle="Rename chat"
  isfooter="true"
>
  <div
    class="chat-actions-dialog-body"
  >
    <div
      class="chat-rename-field"
    >
      <span
        appearance="heading-md"
      >
        * indicates required field
      </span>
      <textarea
        data-testid="chat-name-field"
        label="Name*"
        maxlength="128"
      >
        fakeChat
      </textarea>
    </div>
  </div>
  <div
    class="delete-dialog-footer"
    slot="footer"
  >
    <button
      appearance="secondary"
      data-testid="chat-dialog-cancel"
    >
      Cancel
    </button>
    <button
      appearance="primary"
      data-testid="chat-dialog-action"
    >
      Rename chat
    </button>
  </div>
</div>
`;
